games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+  private resetHandsMove(player: RummyPlayer) {
+    if (player.playerType === 'house') {
+      player.hand.forEach((card, i) => {
+        this.time.delayedCall(1000, () => {
+          card.playMoveTween(
+            this.playerHandZones[1].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              i * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[1].y - 50
+          );
+        });
+      });
+    } else {
+      // player
+      player.hand.forEach((card, i) => {
+        this.time.delayedCall(1000, () => {
+          card.playMoveTween(
+            this.playerHandZones[0].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              i * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[0].y + 50
+          );
+        });
+      });
+    }
+  }
+
+  private async cardPutOnDeckFront(player: RummyPlayer) {
+    if (player.playerType === 'player') {
+      this.enableCardDraggable();
+      console.log('select card to deck!!');
+      // dragの移動とhandからの削除まとめて
+      await this.playerHandSelectedCard();
+      this.disableCardDraggable();
+      this.resetHandsMove(this.players[0]);
+
+      await this.delay(2000);
+      // eslint-disable-next-line no-else-return
+    } else {
+      // house
+      const randomIndex = Math.floor(
+        Math.random() * this.players[1].hand.length
+      );
+      const discardedCard =
+        this.players[1].hand[randomIndex];
+      if (discardedCard) {
+        this.children.bringToTop(discardedCard);
+        discardedCard.playMoveTween(
+          this.deckFrontCardX,
+          this.deckFrontCardY
+        );
+        discardedCard.playFlipOverTween();
+        this.deckFront.push(discardedCard);
+        player.removeCardFromHand(discardedCard);
+      }
+    }
+  }
+
+  private playerHandSelectedCard(): Promise<void> {
+    return new Promise<void>((resolve) => {
+      this.input.setDraggable(this.players[0].hand);
+      this.createDropZoneEvent().then(() => {
+        this.input.setDraggable(
+          this.players[0].hand,
+          false
+        );
+        resolve();
+      });
+    });
+  }
+
+  private checkIfCardOverDeck(card: Card) {
+    const deckBounds = this.deckFront[0].getBounds();
+    const cardBounds = card.getBounds();
+
+    return Phaser.Geom.Intersects.RectangleToRectangle(
+      deckBounds,
+      cardBounds
+    );
+  }
+
+  // メルドゾーンにおくカードを選定
+  // eslint-disable-next-line class-methods-use-this
+  private findMeldAbleCard(hands: Card[]) {
+    const meldAbleCards: Card[][][] = [[], []];
+
+    // 同じrankが3枚以上ある場合の処理
+    const rankCounts: Record<string, number> = {};
+    hands.forEach((card) => {
+      if (rankCounts[card.rank]) rankCounts[card.rank] += 1;
+      else rankCounts[card.rank] = 1;
+    });
+
+    Object.keys(rankCounts).forEach((rank) => {
+      if (rankCounts[rank] >= 3) {
+        const cards = hands.filter(
+          (card) => card.rank.toString() === rank.toString()
```

Comment:
rankは元々stringなので、toString()は不要です。

```suggestion
          (card) => card.rank === rank
```

Reply:
toString()を外すと意図通りに動作しないらしい。自環境で確認する。